zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -555,7 +566,11 @@ angular.module('zeppelinWebApp')
     if (isNaN(timeMs) || timeMs < 0) {
       return '&nbsp;';
     }
-    return 'Took ' + (timeMs/1000) + ' seconds';
+    var desc = 'Took ' + (timeMs/1000) + ' seconds.';
+    if (pdata.textSaved !==undefined && Date.parse(pdata.textSaved) > Date.parse(pdata.dateStarted)){
+      desc += ' (outdated)'
+    }
+    return desc;
```

Comment:
in this flow, should it show the (outdated) text even if it has never been run (ie. timeMs NaN or 0)? because it is returning above at line 567